subsys/bluetooth/controller/ll_sw/ctrl.c

```diff
@@ -1095,7 +1105,71 @@ static u32_t isr_rx_scan_report(u8_t rssi_ready)
 	return 0;
 }
 
-static inline u32_t isr_rx_scan(u8_t irkmatch_id, u8_t rssi_ready)
+static inline bool isr_rx_scan_check(u8_t irkmatch_ok, u8_t devmatch_ok,
+				     u8_t rl_idx)
+{
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_PRIVACY)
+	return ((((_radio.scanner.filter_policy & 0x01) == 0) &&
+		(!devmatch_ok || ctrl_rl_idx_allowed(irkmatch_ok, rl_idx))) ||
```

Comment:
(!devmatch_ok && ctrl_rl_idx_allowed) : not in the resolving list or allowed because it's in device privacy mode or it has no irk, etc

(devmatch_ok && ctrl_irk_whitelisted(rl_idx) : in the whitelist or IRK whitelisted